FirebaseInstallations/Source/Library/FIRInstallationsErrorUtil.m

```diff
@@ -44,4 +44,31 @@ + (NSError *)installationItemNotFoundForAppID:(NSString *)appID appName:(NSStrin
                          }];
 }
 
++ (NSError *)apiErrorWithHTTPCode:(NSUInteger)HTTPCode {
+  // TODO: Form a proper error.
+  NSString *failureReason = [NSString
+      stringWithFormat:@"Unexpected server response HTTP code: %lu", (unsigned long)HTTPCode];
+  return [NSError errorWithDomain:@"FIRInstallationsError"
+                             code:-1
+                         userInfo:@{
+                           NSLocalizedFailureReasonErrorKey : failureReason,
+                         }];
+}
+
++ (NSError *)JSONSerializationError:(NSError *)error {
+  // TODO: Form a proper error.
+  return error;
+}
+
++ (NSError *)FIDRegestrationErrorWithResponseMissingField:(NSString *)missingFieldName {
+  // TODO: Form a proper error.
```

Comment:
Do we need to create a helper for a specific error function like this? Just want to check if it can be merged with above apiErrorWithHTTTPCode.  It's worth of creating a helper function only if it's been called over many times for simplicity reason, and the old IID SDK is not a good example :P

Reply:
@chliangGoogle  I would keep this error type as a separate one. It is different from a wrong HTTP status code error and provides information on an unexpected response format. I am planning to add such errors as `NSUnderlyingErrorKey` the SDK public error to provide additional information for us when we investigate potential issues. Though, I think, I'll rename it to a more generic like `FIDAPIResponseParsingErrorWithMissingField` later on. 
WDYT?

Reply:
is it going to be a lot of missing field errors?

Reply:
So far the error is used in 4 places. I hope, we will never have this error happening in production though :)